dotnet/src/Connectors/Connectors.Memory.Postgres/Database.cs

```diff
@@ -107,16 +132,12 @@ public async Task CreateIndexAsync(NpgsqlConnection conn, CancellationToken canc
     /// <returns></returns>
     public async Task CreateCollectionAsync(NpgsqlConnection conn, string collectionName, CancellationToken cancellationToken = default)
     {
-        if (await this.DoesCollectionExistsAsync(conn, collectionName, cancellationToken).ConfigureAwait(false))
-        {
-            // Collection already exists
-            return;
-        }
-
         using NpgsqlCommand cmd = conn.CreateCommand();
         cmd.CommandText = $@"
             INSERT INTO {TableName} (collection, key)
-            VALUES(@collection, '')";
+            VALUES(@collection, '')
```

Comment:
What is the purpose of adding collection with empty key? Isn't it just an empty row?

Reply:
Because `collection` and `key` are joint primary keys, they cannot be null.

Reply:
Correct, but since this implementation uses only one table (which is mentioned in my previous comment), what is the point in adding an empty row? If I use `UpsertAsync` without `CreateCollectionAsync` first, my upsert operation will succeed, is that correct?

Reply:
Yes, records can be upsert even if the collection does not exist. Also referring to Sqlite 😅, I think it is designed to fetch already created collections when the user only creates a collection and does not insert any records.

Reply:
We will improve Sqlite if needed as well :)
That's why I posted my question about table-collection mapping. In case we create new table per collection, it will resolve such problems for us.

Current implementation is fine for demos and examples, but I'm not sure it will be possible to use it in production.